scenarios/register.go

```diff
@@ -0,0 +1,7 @@
+package scenarios
+
+// RegisterScenarios registers all scenarios. It is intended to be called by Go
+// applications using omes as a library to run scenarios.
+func RegisterScenarios() {
```

Comment:
Can this method be removed? If including a Go package has side effects, it can be documented at the package level and users can import it for those side effects.

Reply:
So, the options (as seen from the library user's point of view) are:

### A
```go
import (
    "github.com/temporalio/omes/scenarios"
)
scenarios.RegisterScenarios()
```

and

### B
```go
import (
    _ "github.com/temporalio/omes/scenarios" // register scenarios by side-effect
)
```

IMO **A** is better library design for two reasons:
1. It's natural and explicit
2. It hides the implementation detail and allows us to change the way that scenarios are registered in the future. I.e., the fact that scenario registration happens as a side-effect of import should be considered an implementation detail of the library; we don't want changing that to be a breaking change for users.

Reply:
> IMO A is better library design for two reasons:

But it's not really about personal preference on these Go things. In Go, imports with side effects are usually done via a side-effecting import, not a no-op method. (even if it was a method, it would be named something that it doesn't do, "register" is inaccurate because it is done whether you call the method or not)

But I would totally support changing the scenarios to not register on `init()` but instead only register in `RegisterScenarios` and just use `init()` to build up a slice of them to be registered in that function. Just don't forget to call `RegisterScenarios` for the CLI too.

Reply:
> But it's not really about personal preference on these Go things. In Go, imports with side effects are usually done via a side-effecting import, not a no-op method. (even if it was a method, it would be named something that it doesn't do, "register" is inaccurate because it is done whether you call the method or not)

I think you're misunderstanding slightly -- all that is deliberate. My proposal here was to deliberately introduce a no-op method that says it is registering scenarios when in fact it is doing nothing and the scenarios were registered already. `RegisterScenarios` was an accurate name in a forwards-compatibility sense because it was papering over a bug in the library that should be fixed in the future -- i.e. the registration should be moved into the function.

The reason I was proposing that we do this is that a library should never require users of the library to perform an import for side-effects only. It's fine to use Go's import side effect feature internally, but a library telling users that, in order to use the library, they must import something and not use it, is something that should be avoided.

That said, I agree that it would be better to make the public interface match up with the implementation, instead of going through a transition period in which a public function lies about what it is doing. Furthermore there's no real need to address this now, since the only known user of omes as a library is Temporal's CI/CD framework, which is written by the authors of omes. Or put another way, this defect in omes-as-a-library has been released to the public for a while and there's nothing about current work on internal CI/CD that makes it urgent to address the library design issue. So I'll revert this change and merge the PR with the random docstring/README edits only.

Reply:
:+1: Definitely don't want a no-op for something that an import does. If we need to move from side-effecting import to function-based registration we can (and if you want to wait until some future time, no prob)

> It's fine to use Go's import side effect feature internally, but a library telling users that, in order to use the library, they must import something and not use it, is something that should be avoided.

This is not necessarily true in Go. Quite often side-effecting imports are used in libraries (e.g. k8s, protobuf, etc). Having said that, I have no problem with the more explicit approach.

> this defect in omes-as-a-library has been released to the public

Not really a "defect" and we should not have any backwards compatibility guarantees in omes libraries in any way. Omes should be able to be completely changed/refactored at will to whatever is best as needs arise.

> So I'll revert this change and merge the PR with the random docstring/README edits only.

:+1: (though it should be super easy to just move to explicit registration)